src/Security/MemberAuthenticator/Authenticator.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace SilverStripe\Security\MemberAuthenticator;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\Session;
+use SilverStripe\ORM\ValidationResult;
+use InvalidArgumentException;
+use SilverStripe\Security\Authenticator as BaseAuthenticator;
+use SilverStripe\Security\Security;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\LoginAttempt;
+
+/**
+ * Authenticator for the default "member" method
+ *
+ * @author Markus Lanthaler <markus@silverstripe.com>
+ */
+class Authenticator implements BaseAuthenticator
+{
+
+    public function supportedServices()
+    {
+        // Bitwise-OR of all the supported services, to make a bitmask
+        return BaseAuthenticator::LOGIN | BaseAuthenticator::LOGOUT | BaseAuthenticator::CHANGE_PASSWORD
+            | BaseAuthenticator::RESET_PASSWORD;
+    }
+
+    /**
+     * @inherit
+     */
+    public function authenticate($data, &$message)
+    {
+        $success = null;
+
+        // Find authenticated member
+        $member = $this->authenticateMember($data, $message, $success);
+
+        // Optionally record every login attempt as a {@link LoginAttempt} object
+        $this->recordLoginAttempt($data, $member, $success);
+
+        if ($member) {
+            Session::clear('BackURL');
+        }
+
+        return $success ? $member : null;
+    }
+
+    /**
+     * Attempt to find and authenticate member if possible from the given data
+     *
+     * @param array $data Form submitted data
+     * @param $message
```

Comment:
It would be a good idea to do so. I think it would improve usability a lot.

There's only one but: I discussed this with @chillu a bit today, and we're fighting a tight deadline here.
I would be happy to implement these changes to use a ValidationResult, but if it would delay this PR to not reach beta-1, I would prefer to drop it for now?